src/core/tsi/transport_security.c

```diff
@@ -330,3 +331,50 @@ tsi_result tsi_construct_peer(size_t property_count, tsi_peer *peer) {
   }
   return TSI_OK;
 }
+
+/* --- tsi_plugin_func implementation --- */
+
+typedef struct tsi_plugin_func_list tsi_plugin_func_list;
+
+static tsi_plugin_func_list *tsi_destroy_list = NULL;
+static gpr_once tsi_once_var = GPR_ONCE_INIT;
+static gpr_mu tsi_mu;
```

Comment:
Same here -- as long as we require that `tsi_register_destroy_plugin()` is only invoked at initialization time, there should be no need for locking.